### PR TITLE
Collapsible doc sections

### PR DESCRIPTION
**Pull request by DoodleBobBuffPants**

### CODE REVIEW

**Comment by Lyeeedar** [comment]
Currently it collapses whenever you change page. Could we get it to remember the expansion / collapse status as you transition around the site?

Also if following a link to a non-expanded page, I feel it should expand to show that page.

**Comment by DoodleBobBuffPants** [comment]
Currently it'll work for some and not for all. It sets the `open` prop on `<details></details>` tags based on the URL, so for databases for example we can tell whether the section should be open by checking the URL, because there isn't any built in support for this. Other sections don't have a common URL prefix to tell them apart, so we'd either need to refactor the pages or add a Liquid variable for each section

**Comment by Lyeeedar** [comment]
https://idratherbewriting.com/documentation-theme-jekyll/mydoc_sidebar_navigation.html worth investigating?

**Comment by DoodleBobBuffPants** [comment]
> https://idratherbewriting.com/documentation-theme-jekyll/mydoc_sidebar_navigation.html worth investigating?

It seems like it's a website template instead of an actual theme

File: documentation/usage/maven/index.md [hunk 1]
```diff
@@ -205,8 +205,8 @@ Configuration can also be supplied directly via the command-line using JVM syste
 
 ### Config files
 
-[Config files](/documentation/configfiles) are supported by the Flyway Maven plugin. If you are not familiar with them,
-check out the [Flyway config file structure and settings reference](/documentation/configfiles) first.
+[Config files](/documentation/configuration/configfiles) are supported by the Flyway Maven plugin. If you are not familiar with them,
```
Comment (Lyeeedar) [inline documentation/usage/maven/index.md:209]: All these links to the `configfiles` page should be `configFiles` (might have been originally broken by my commit, but should get fixed anyway)

**Review by Lyeeedar — CHANGES_REQUESTED** [review]
Could we change `/documentation/configuration/configuration` to `/documentation/configuration/parameters`?

**Review by Lyeeedar — APPROVED** [review]

**Review by MikielAgutu — CHANGES_REQUESTED** [review]
We have to be careful with broken links.

For instance, `/documentation/configuration/` has moved to `documentation/configuration/parameters/`. However there's no redirect. We need to introduce redirects to preserve old links.

**Comment by DoodleBobBuffPants** [comment]
> We have to be careful with broken links.
> 
> For instance, `/documentation/configuration/` has moved to `documentation/configuration/parameters/`. However there's no redirect. We need to introduce redirects to preserve old links.

Is this referring to broken links within the website? Instead of redirected they should be updated

**Comment by MikielAgutu** [comment]
Sorry, let me clarify.

When I say 'broken links' you can think of it like a breaking change to the 'API' of the website. Except in this case the API is the url.

At the moment `https://flywaydb.org/documentation/configuration/` is a valid location on the website. However, if this change goes in, that same url will link to an invalid page. So that location will be 'broken'.

This is a problem because web links tend to die hard. For instance, when we respond to customers on GitHub we often link to the docs. If those links suddenly stop working that's a problem because people tend to visit such posts years after they're created.

How many times have you visited a Stack Overflow post that happens to be years old? Wouldn't it be annoying if the +100 verified answer was link to a docs page that no longer exists?

The solution is to introduce sensible redirects for locations that have been deprecated. In this case `documentation/configuration/` ought to be redirected to `documentation/configuration/parameters/` since that's the equivalent page. Jekyll does support redirects, though I think you need to use a plugin (maybe https://github.com/jekyll/jekyll-redirect-from).

**Review by Lyeeedar — CHANGES_REQUESTED** [review]
- Currently if you select the 'errorOverrides', 'callbacks', or 'dry runs' pages from either `get started/advanced` or `concepts`, both pages in the menu show as selected, suggesting they are the same page (even though they arent).

  - On that topic, we should probably link from the concepts pages to the get started/advanced ones (under a `see example tutorial` or `learn more about this`) as the tutorials seems pretty useful + give good use cases.

- `Config files` link in `configuration` is broken.

- Sub menu padding is really high + the lack of background makes it look a little messy. Could we reduce the padding / add background to the submenus?

**Comment by MikielAgutu** [comment]
image
Sub headings are in line with the top level heading

**Review by MikielAgutu — APPROVED** [review]
🚀

**Review by Lyeeedar — APPROVED** [review]

**Pull request merged**